plugins/org.eclipse.fordiac.ide.deployment/src/org/eclipse/fordiac/ide/deployment/data/ResourceDeploymentData.java

```diff
@@ -137,18 +139,72 @@ public static String getMappedParentName(final FBNetworkElement fbnElement) {
 
 	private void addSubAppParams(final SubApp subApp, final Deque<SubApp> subAppHierarchy, final String prefix)
 			throws DeploymentException {
+		// follow the connection through the subapp hierachy until there is a loose one
 		for (final VarDeclaration dataInput : subApp.getInterface().getInputVars()) {
-			final Function<VarDeclaration, String> dataInputValue = DeploymentHelper
-					.getVariableValueRetargetable(dataInput, subApp);
-			if (dataInputValue != null) {
-				for (final ConDeploymentDest destData : getSubappInterfaceconnections(subAppHierarchy, prefix,
-						dataInput)) {
-					final VarDeclaration destVar = (VarDeclaration) destData.destination;
-					final String destDataValue = dataInputValue.apply(destVar);
-					params.add(new ParameterData(destDataValue, destData.prefix, destVar));
+
+			if (useSubAppInitialValue(subAppHierarchy, subApp, dataInput)) {
```

Comment:
as the useSubAppInitialValue is an expensive operation I would only call it when necessary. Therefore I would first perform the check `if(dataInputValue != null && useSubappInitialValue(..))` 
Otherwise deployment may be tremendously slower.